go/libraries/utils/argparser/parser.go

```diff
@@ -271,6 +271,33 @@ func (ap *ArgParser) matchValueOption(arg string, isLongFormFlag bool) (match *O
 	return nil, nil
 }
 
+func (ap *ArgParser) ParseGlobalArgs(args []string) (apr *ArgParseResults, remaining []string, err error) {
+	list := make([]string, 0, 16)
+	results := make(map[string]string)
+
+	i := 0
+	for ; i < len(args); i++ {
+		arg := args[i]
+
+		if len(arg) == 0 || arg == "--" {
+			continue
+		}
+
+		if arg[0] != '-' {
```

Comment:
Probably worth a one line comment here that no subcommand starts with '-', and that we assume this is a valid subcommand.

I don't see anyplace were we reject totally bogus commands - or having no command at all. Could remaining list be len(0) and line 531 in dolt.go result in a out of range error?

Reply:
If there's no subcommand we return an error on line 299.

`remaining` always contains at least the subcommand, so out of range is not possible there.

I added a comment.

Reply:
Huge improvement! Bravo!